rust-semantics/expression/struct.md

```diff
@@ -5,8 +5,89 @@ module RUST-EXPRESSION-STRUCT
     imports private RUST-CONVERSIONS-SYNTAX
     imports private RUST-ERROR-SYNTAX
     imports private RUST-EXECUTION-CONFIGURATION
+    imports private RUST-PREPROCESSING-CONFIGURATION
     imports private RUST-REPRESENTATION
 
+    // For freezing heating operations, when leaving a struct alone in the topcell, make it a ptrValue
+    rule <k> struct(P, Fields) => ptrValue(null, struct(P, Fields)) ... </k>
+
+    // KItems for breaking down struct expressions into the adequate format struct(P, F)
+    syntax KItem ::= fromStructExpressionWithLiteralsBuildFieldsMap(TypePath, StructBases, List, Map)
```

Comment:
Probably it's a good idea to not implement this, since we are short on time, but, for the future, probably the better way of implementing this is something like this:
```
syntax KItem ::= fromStructExpressionWithLiteralsBuildFieldsMap(TypePath, StructBases, List)  [strict(2), result(ValueWithPtr)]

rule isValueWithPtr(.StructBases) => true
rule isValueWithPtr(E:Expression Es:StructBases) => isValueWithPtr(E) andBool isValueWithPtr(Es)

rule E:Expression Es:StructBases => E ~> E Es  [heat, result(ValueWithPtr)]
rule E:Expression ~> _:Expression Es:StructBases => E Es  [cool, result(ValueWithPtr)]

rule E:Expression Es:StructBases => Es ~> E Es requires isValueWithPtr(E)  [heat, result(ValueWithPtr)]
rule Es:StructBases ~> E:Expression _:StructBases => E Es requires isValueWithPtr(E)  [cool, result(ValueWithPtr)]

rule fromStructExpressionWithLiteralsBuildFieldsMap(I, RL, FieldNameList) => almostAStruct(I, pairFieldsWithValues(RL, FieldNameList, .Map))
rule almostAStruct(I, M:Map) => struct(I, M)

syntax MapOrError ::= pairFieldsWithValues(StructBases, List, .Map)  [function, total]
rule pairFieldsWithValues(.StructBases, .List, M:Map) => M
rule pairFieldsWithValues(???) => ???
...
```

Something similar applies to `fromStructExpressionWithAssignmentsBuildFieldsMap`.